cmd/spicedb/serve.go

```diff
@@ -275,51 +278,30 @@ func serveRun(cmd *cobra.Command, args []string) {
 	}
 
 	redispatch := graph.NewLocalOnlyDispatcher(nsm, ds)
-	redispatchClientCtx, redispatchClientCancel := context.WithCancel(context.Background())
-
-	redispatchTarget := cobrautil.MustGetStringExpanded(cmd, "dispatch-cluster-dns-name")
-	redispatchServiceName := cobrautil.MustGetStringExpanded(cmd, "dispatch-cluster-service-name")
-	if redispatchTarget != "" {
-		log.Info().Str("target", redispatchTarget).Msg("initializing remote redispatcher")
-
-		resolverAddr := cobrautil.MustGetStringExpanded(cmd, "dispatch-peer-resolver-addr")
-		resolverCertPath := cobrautil.MustGetStringExpanded(cmd, "dispatch-peer-resolver-tls-cert-path")
-		var resolverConfig *consistentbackend.EndpointResolverConfig
-		if resolverCertPath != "" {
-			log.Debug().Str("addr", resolverAddr).Str("cacert", resolverCertPath).Msg("using TLS protected peer resolver")
-			resolverConfig = consistentbackend.NewEndpointResolver(resolverAddr, resolverCertPath)
-		} else {
-			log.Debug().Str("addr", resolverAddr).Msg("using insecure peer resolver")
-			resolverConfig = consistentbackend.NewEndpointResolverNoTLS(resolverAddr)
-		}
+
+	// grpc consistent loadbalancer redispatch configuration
+	dispatchAddr := cobrautil.MustGetStringExpanded(cmd, "dispatch-upstream-addr")
+	if len(dispatchAddr) > 0 {
+		log.Info().Str("upstream", dispatchAddr).Msg("configuring grpc consistent load balancer for redispatch")
 
 		peerPSK := cobrautil.MustGetStringExpanded(cmd, "grpc-preshared-key")
 		peerCertPath := cobrautil.MustGetStringExpanded(cmd, "dispatch-cluster-tls-cert-path")
```

Comment:
We actually want a different config flag for the client certificate, and we want to use the CA as the certificate.

Reply:
I didn't see any client cert config in `main`, just the CA config. The credentials here are configured the same as here: https://github.com/authzed/spicedb/blob/d6dec1252db5065d29d726f3c157462f0ca40fd6/internal/dispatch/client/consistentbackend/options.go#L62

(but I removed the dns name flag, it looked unnecessary to me based on how the `ServerName` field is treated if missing in grpc ClientHandshake: https://github.com/grpc/grpc-go/blob/master/credentials/tls.go#L76-L83

Reply:
Added a new `--dispatch-cluster-upstream-ca-path` flag to do this